### ISSUE TITLE
Discrepancy in training step accuracy with Tensorflow 2.1

### ISSUE THREAD

**Issue by kunal4097** [tensorflow/tensorflow#36366]
tag:Bug/Performance Issue</em>

**Describe the current behavior**
1. I followed each and every step from the tensorflow documentation but still there is discrepancy coming in training step accuracy 
2. Training and validation accuracy is far too different though same dataset is used for training and validation

**Describe the expected behavior**
1. Training accuracy should not come that different from the validation accuracy
2. model.fit and model.evaluate accuracy should be nearly same 
**Code to reproduce the issue**
I have attached the link to my google colab notebook: 
[Google Colab Link](https://colab.research.google.com/drive/1Ir8LhIrYB-UexygQKIFuNwRkmb6yJh-w)
<img width="1095" alt="Screenshot 2020-01-31 at 3 48 50 PM" src="https://user-images.githubusercontent.com/11526652/73531945-931dfb00-4441-11ea-8401-0cca6ea79142.png">

**Comment by amahendrakar** [comment]
Was able to reproduce the issue. Please find the Gist [here](https://colab.research.google.com/gist/amahendrakar/5ac084bdfd001085dba47ef5f1aa1210/copy-of-36366.ipynb). Thanks!

**Comment by raghavab1992** [comment]
I am also facing this issue with both tensorflow 2.0 and 2.1. This is pretty weird. I am not sure if I am missing anything as it looks like a big issue/bug to have if it is true. Can you please help with this asap?

**Comment by peeyush10234** [comment]
Cannot find proper documentation for the same and facing the same issue.

**Comment by raghavab1992** [comment]
@ymodak @amahendrakar  can u please help with this?

**Comment by ymodak** [comment]
Please take a look at tensorflow official example using same flower dataset https://colab.sandbox.google.com/github/tensorflow/examples/blob/master/community/en/flowers_tf_lite.ipynb
Thanks!

**Comment by raghavab1992** [comment]
@ymodak I was able to solve this issue by adding layers argument in the keras model as suggested in this [issue](](https://github.com/keras-team/keras/pull/9965#issuecomment-549126009) link

`tf.keras.applications.mobilenet.MobileNet(input_shape=[224, 224, 3], layers = tf.keras.layers)}`

This issue has been affecting many and is documented very well in this keras [issue thread](https://github.com/keras-team/keras/pull/9965) and [blog](http://blog.datumbox.com/the-batch-normalization-layer-of-keras-is-broken/)

Please fix this issue and update the existing tensorflow2 documentation with this fix asap

**Comment by tomerk** [comment]
The next 2.2 release candidate will include this commit:
https://github.com/tensorflow/tensorflow/commit/410852dbd24899e22f0020f9fdc9757f527dda55
(This fixes a bug where some keras code was accidentally referring to the wrong versions internally)

If you previously had to put `layers=tf.keras.layers`, this should fix your issue.

**Comment by hsd31196** [comment]
> The next 2.2 release candidate will include this commit:
> [410852d](https://github.com/tensorflow/tensorflow/commit/410852dbd24899e22f0020f9fdc9757f527dda55)
> (This fixes a bug where some keras code was accidentally referring to the wrong versions internally)
> 
> If you previously had to put `layers=tf.keras.layers`, this should fix your issue.

I tried adding layers='tf.keras.layers'  but still it is not working for me.

**Comment by ymodak** [comment]
This is fixed now and available with latest tf-nightly version as well as release candidate rc-3. Thanks!

**Comment by Azkarehman** [comment]
> I am also facing this issue with both tensorflow 2.0 and 2.1. This is pretty weird. I am not sure if I am missing anything as it looks like a big issue/bug to have if it is true. Can you please help with this asap?

I also faced the same issue
